api4/user_test.go

```diff
@@ -5882,6 +5919,58 @@ func TestThreadSocketEvents(t *testing.T) {
 		require.Truef(t, caught, "User should have received %s event", model.WebsocketEventThreadReadChanged)
 	})
 
+	// read the thread
+	_, resp, err = th.Client.UpdateThreadReadForUser(th.BasicUser.Id, th.BasicTeam.Id, rpost.Id, replyPost.CreateAt+1)
+	require.NoError(t, err)
+	CheckOKStatus(t, resp)
+
+	t.Run("Listen for thread updated event after create post", func(t *testing.T) {
+		testCases := []struct {
+			post        *model.Post
+			preReplies  float64
+			preMentions float64
+			replies     int64
+			mentions    int64
+		}{
+			{&model.Post{ChannelId: th.BasicChannel.Id, Message: "simple reply", UserId: th.BasicUser2.Id, RootId: rpost.Id}, 0, 0, 1, 0},
+			{&model.Post{ChannelId: th.BasicChannel.Id, Message: "mention reply 1 @" + th.BasicUser.Username, UserId: th.BasicUser2.Id, RootId: rpost.Id}, 1, 0, 2, 1},
+			{&model.Post{ChannelId: th.BasicChannel.Id, Message: "mention reply 2 @" + th.BasicUser.Username, UserId: th.BasicUser2.Id, RootId: rpost.Id}, 2, 1, 3, 2},
+			{&model.Post{ChannelId: th.BasicChannel.Id, Message: "self reply", UserId: th.BasicUser.Id, RootId: rpost.Id}, 3, 2, 0, 0}, // posting as current user will read the thread
+			{&model.Post{ChannelId: th.BasicChannel.Id, Message: "simple reply", UserId: th.BasicUser2.Id, RootId: rpost.Id}, 0, 0, 1, 0},
+			{&model.Post{ChannelId: th.BasicChannel.Id, Message: "mention reply 3 @" + th.BasicUser.Username, UserId: th.BasicUser2.Id, RootId: rpost.Id}, 1, 0, 2, 1},
+		}
+
+		for _, tc := range testCases {
+			// post a reply on the thread
+			_, appErr = th.App.CreatePostAsUser(th.Context, tc.post, th.Context.Session().Id, false)
+			require.Nil(t, appErr)
+
+			var caught bool
+			func() {
+				for {
+					select {
+					case ev := <-userWSClient.EventChannel:
+						if ev.EventType() == model.WebsocketEventThreadUpdated {
+							caught = true
+							data := ev.GetData()
+							var thread model.ThreadResponse
+							jsonErr := json.Unmarshal([]byte(data["thread"].(string)), &thread)
```

Comment:
Is there a need for this double-conversion? Could we directly cast to `[]byte`?

Reply:
It asks for type assertion, `cannot convert data["thread"] (type interface {}) to type []byte: need type assertion`.

Reply:
You are right, the underlying value is a string. My bad there.